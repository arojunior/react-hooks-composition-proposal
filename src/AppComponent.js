import React from 'react';
import PropTypes from 'prop-types';
import logo from './assets/image/logo.svg';
import './assets/css/App.css';

const AppComponent = ({ useFoo, useGithub }) => {
  const { foo, changeFoo } = useFoo("bar");
  const { user } = useGithub("arojunior");
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <h2>Hello {foo}</h2>
        <h2>Start editing to see some magic happen!</h2>
        <button onClick={() => changeFoo("wooow")}>Change bar</button>
        <div>
          <p>
            <strong>Name: </strong>
            {user.name}
          </p>
        </div>
      </header>
    </div>
  );
};

AppComponent.propTypes = {
  useFoo: PropTypes.func.isRequired,
  useGithub: PropTypes.func.isRequired
};

export default AppComponent;
