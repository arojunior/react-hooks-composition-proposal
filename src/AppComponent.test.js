import React from "react";
import renderer from "react-test-renderer";
import AppComponent from "./AppComponent";

describe("AppComponent", () => {
  test("should match snapshot", () => {
    const useFoo = jest.fn(() => ({}));
    const useGithub = jest.fn(() => ({ user: {} }));

    const tree = renderer
      .create(<AppComponent useFoo={useFoo} useGithub={useGithub} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
