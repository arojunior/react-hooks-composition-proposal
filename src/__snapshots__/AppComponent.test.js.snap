// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Hello 
    </h2>
    <h2>
      Start editing to see some magic happen!
    </h2>
    <button
      onClick={[Function]}
    >
      Change bar
    </button>
    <div>
      <p>
        <strong>
          Name: 
        </strong>
      </p>
    </div>
  </header>
</div>
`;
